src/mappings/serviceAgreement.ts

```diff
@@ -15,8 +15,10 @@ export async function handleServiceAgreementCreated(
   logger.info('handleClosedServiceAgreementCreated');
   assert(event.args, 'No event args');
 
+  const eventServiceAgreementId = event.args.serviceAgreementId.toString();
```

Comment:
Should this be event.args.serviceAgreementId.toHexString() in order to connect with factory?

Reply:
No factory need anymore, I mention this before that we don't create agreement as new contract. You can access the agreement information with the following code:

```
const agreementRegistry = IServiceAgreementRegistry__factory.connect(
    eventServiceAgreementId,
    new FrontierEthProvider()
  );
  
const agreement = await agreementRegistry.closedServiceAgreements(agreementId)
const { period, lockedAmount } = agreement;
```

Reply:
updated